### DISCUSSION TITLE
How to get listings and NFT details in the same request

### DISCUSSION THREAD

**Discussion by aaisataev** [ProjectOpenSea/opensea-js#1323] [Q&A]
Hi,

I have a use case of showing listings and NFT details in the same page(name, price, attributes).
With the existing API I found the only way to fetch listings first then get NFTs after. But this takes N+1 requests for N items which is blocked by the rate limiter.

Any other ways to fetch listings with NFT details?

Best,
AI

**Comment by ryanio** [comment]
You can get listings with the nft details returned in the same call with the events API
